src/util-inl.h

```diff
@@ -234,27 +234,39 @@ bool StringEqualNoCaseN(const char* a, const char* b, size_t length) {
 // compiler version specific functionality.
 // malloc(0) and realloc(ptr, 0) have implementation-defined behavior in
 // that the standard allows them to either return a unique pointer or a
-// nullptr for zero-sized allocation requests.  Normalize by always using
-// a nullptr.
+// nullptr for zero-sized allocation requests. Normalize by always using
+// a static buffer.
+static void* fake_mem = malloc(1);
```

Comment:
@mscdex Yeah, that should work, but it may come with a bit of overhead and it’s not technically thread-safe.

Something that might be a bit faster would be using a compile-time static buffer, i.e. `extern char fake_mem[1];` in `util-inl.h` and `char fake_mem[1];` in `util.cc`, because that should enable the compiler to inline the memory address (which may become an actual constant).